src/net_processing.cpp

```diff
@@ -779,6 +779,21 @@ void PeerLogicValidation::InitializeNode(CNode *pnode) {
         PushNodeVersion(pnode, connman, GetTime());
 }
 
+// Note: unbroadcast transactions will be filtered through filterInventoryKnown
+void PeerLogicValidation::ReattemptInitialBroadcast(CScheduler& scheduler) const
+{
+    std::set<uint256> unbroadcast_txids;
+    WITH_LOCK(m_mempool.cs, unbroadcast_txids = m_mempool.m_unbroadcast_txids);
+
+    for (const uint256& txid : unbroadcast_txids) {
+        RelayTransaction(txid, *connman);
+    }
+
+    // schedule next run for 10-15 minutes in the future
+    const std::chrono::milliseconds delta = std::chrono::milliseconds(10 * 60 * 1000 + GetRand(5 * 60 * 1000));
```

Comment:
```suggestion
    const std::chrono::milliseconds delta = std::chrono::minutes{10} + GetRandMillis(std::chrono::minutes{5});
```

You might have to add `GetRandMillis` by copying from `GetRandMicros`.

Reply:
added the method `GetRandMillis` & updated here. I was thinking might make sense to have a template for the different chrono types? I haven't looked into it yet but you mentioned that it wouldn't work?

Reply:
@amitiuttarwar Done here: #18751